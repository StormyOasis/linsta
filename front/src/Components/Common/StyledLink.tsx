import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { Div } from "./CombinedStyling";

export type StyledLinkProps = {
    to?: string;
    children?: any;
    onClick?: any;
    styleOverride?: any;
    className?: any;
    datatestid?: string
};

const StyledLinkWrapperFromLink = styled(Link)`
    color: ${props => props.theme['colors'].buttonDefaultColor};
    text-decoration: none;
    font-weight: 600;
    cursor: pointer;

    &:hover {
        color: ${(props) => props.theme['colors'].buttonOnHoverColor};
    }
`;

const StyledLinkWrapper = styled(Div)`
    color: ${props => props.theme['colors'].buttonDefaultColor};
    text-decoration: none;
    font-weight: 600;
    cursor: pointer;

    &:hover {
        color: ${(props) => props.theme['colors'].buttonOnHoverColor};
    }
`;

const StyledLink: React.FC<StyledLinkProps> = (props: StyledLinkProps) => {
    const onClick = props.onClick ? props.onClick : () => true;
    const hasTo: boolean = (props.to != null && props.to.length > 0);

    return (
        <>
            {hasTo && (
                <StyledLinkWrapperFromLink data-testid={props.datatestid} to={props.to || "#"} onClick={onClick} className={props.className} style={props.styleOverride}>
                    {props.children}
                </StyledLinkWrapperFromLink>
            )}
            {!hasTo && (
                <StyledLinkWrapper data-testid={props.datatestid} onClick={onClick} className={props.className} style={props.styleOverride}>
                    {props.children}
                </StyledLinkWrapper>
            )}
        </>
    );
};

export default StyledLink;