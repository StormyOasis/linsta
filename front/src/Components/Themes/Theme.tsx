import React from "react";
import { ThemeProvider } from "styled-components";

const defaultTheme = {
    colors: {
        defaultLinkColor: "black",
        defaultTextColor: "black",
        backgroundColor: "white",
        backgroundDisabled: "rgba(0, 0, 0, .6)",
        borderDefaultColor: "rgb(220, 220, 220)",
        borderDarkColor: "rgb(180,180, 180)",
        inputBackgroundColor: "rgb(250, 250, 250)",
        navLinkHoverColor: "rgb(236, 236, 236)",
        buttonTextColorDefault: "white",
        buttonSecondaryTextColorDefault: "black",
        buttonDefaultColor: "rgb(0, 150, 245)",
        buttonDefaultSecondaryColor: "rgb(240, 240, 240)",
        buttonDefaultColorTrans: "rgba(0, 150, 245, .5)",
        buttonSecondaryDefaultColorTrans: "rgba(240, 240, 240, .5)",
        buttonOnHoverColor: "rgb(25, 120, 240)",        
        buttonSecondaryOnHoverColor: "rgb(220, 220, 220)",        
        inputTextColor: "rgb(115,115,115)",
        navLinkTextColor: "rgb(0, 50, 100)",
        cropperAspectBkgnd: "rgba(25,25,25,.6)",
        cropperAspectBkgndNoTrans: "rgb(25,25,25)",
        mediumTextColor: "rgb(120, 120, 120)",  
        mediaSliderButtonColor: "rgba(0,0,0, .6)",
        mediaSliderButtonBkgndColor: "rgba(225,225,225, .4)",
        warningTextColor: "rgb(235, 70, 90)"
    },    

    input: {
        backgroundColor: "#EFEFEF",        
    },

    sizes: {
        sideBarNavWidthDefault: "244px",
        sideBarNavWidthNarrow: "72px",
        sideBarNavBottomHeight: "50px",
        cropperHeight: "275px",
        defaultModalWidth: "600px",
        maxModalWidth: "75vw",
        maxModalHeight: "75vh",
        minModalHeight: "300px",
        minPostTextEditorHeight: "140px",
        maxPostTextEditorHeight: "140px",
        feedPostMinWidth: "400px",
        maxCommentModalContentHeight: "75vh",
        minCommentModalContentHeight: "300px",
    },

    breakpoints: {
        sm: "576",
        md: "768",
        lg: "1280"
    }
};

export const LexicalEditorTheme = {
    ltr: 'ltr',
    rtl: 'rtl',
};

const Theme = ({children}: any) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
};

export default Theme;
