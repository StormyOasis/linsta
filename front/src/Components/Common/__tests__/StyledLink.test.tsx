import React from 'react'
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom';
import Jest from 'jest';
import StyledLink from '../StyledLink';
import Theme from '../../Themes/Theme';
import { MemoryRouter } from 'react-router-dom';


test("StyledLink Component", () => {
    const r = render(
        <MemoryRouter>
            <Theme>
                <StyledLink to="/" onClick={() => true}>Hello</StyledLink>
            </Theme>
        </MemoryRouter>).asFragment();

    expect(r).toMatchSnapshot();
});

test("StyledLink Component With Style Override", () => {
    const styleOverride = { color: "red" };
    const r = render(
        <MemoryRouter>
            <Theme>
                <StyledLink to="/" onClick={() => true} styleOverride={styleOverride}>Hello</StyledLink>
            </Theme>
        </MemoryRouter>).asFragment();

    expect(r).toMatchSnapshot();
})


test("StyledLink Component no onclick", () => {
    const r = render(
        <MemoryRouter>
            <Theme>
                <StyledLink>Hello</StyledLink>
            </Theme>
        </MemoryRouter>).asFragment();

    expect(r).toMatchSnapshot();
})