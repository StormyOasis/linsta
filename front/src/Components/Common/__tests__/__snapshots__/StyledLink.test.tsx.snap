// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink Component 1`] = `
<DocumentFragment>
  <a
    class="StyledLink__StyledLinkWrapper-sc-1rgr01q-0 dLJqDG"
    href="/"
  >
    Hello
  </a>
</DocumentFragment>
`;

exports[`StyledLink Component With Style Override 1`] = `
<DocumentFragment>
  <a
    class="StyledLink__StyledLinkWrapper-sc-1rgr01q-0 dLJqDG [object Object]"
    href="/"
  >
    Hello
  </a>
</DocumentFragment>
`;
